[IDE] Add "Add Reference" under project context menu

ketttY opened the pull request:
Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/947128/

Fixes VSTS #974919  - Change "Add" to "Add Reference," Remove "Add" in front of items in Add menu 
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/974919

Fixes VSTS #975256 - Change title of window from "Edit References" to "References"
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/975256

Screenshots:

<img width="306" alt="AddSubMenu" src="https://user-images.githubusercontent.com/372361/64247006-d8c3af80-cf05-11e9-9bb0-b28f3d8a9b93.png">
<img width="147" alt="ReferencesContextMenu" src="https://user-images.githubusercontent.com/372361/64247013-dc573680-cf05-11e9-9dbf-9ea8f5eda63c.png">
<img width="167" alt="AddReferencesDependenciesMenu" src="https://user-images.githubusercontent.com/372361/64247054-f09b3380-cf05-11e9-865e-f2b79f9ed9bc.png">
<img width="444" alt="ReferencesDialog" src="https://user-images.githubusercontent.com/372361/64247018-dfeabd80-cf05-11e9-8481-48ca129dfbdc.png">
<img width="137" alt="ProjectMainMenu" src="https://user-images.githubusercontent.com/372361/64247028-e1b48100-cf05-11e9-92a3-029706e00ed0.png">

ketttY committed:
[IDE] Add "Add Reference" under project context menu

Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/947128/

ketttY commented:
<img width="230" alt="Screen Shot 2019-08-13 at 2 00 26 PM" src="https://user-images.githubusercontent.com/34032260/62966095-52b7bb80-bdd4-11e9-88b7-557c4b888571.png">
This is the current location of this command, please let me know if you think it will be better to put it somewhere else :)

ketttY committed:
Change the position of the command and update the condition to just "Project"

ketttY commented:
<img width="230" alt="Screen Shot 2019-08-13 at 2 00 26 PM" src="https://user-images.githubusercontent.com/34032260/62970298-2a808a80-bddd-11e9-8e62-1a099a7c8e62.png">
This is the new layout.

bratsche reviewed (approved):
👍

mrward reviewed (approved):
I sort of wonder how that `insertbefore="MonoDevelop.WebReferences.WebReferenceCommands.Add"` in the NuGet addin is being used since the screenshot looks good to me - however this insertbefore would suggest a different order. I am guessing the web references addin is loaded later after the NuGet addin.

Not suggesting we look at that here though 😄

jstedfast commented:
GitHub has been stalling out all day yesterday afternoon and this morning for me so I can't review/approve this PR (little blue progress bar gets 90% of the way across my screen and then just stops when I click the "Add your review" button), so I'll just comment here...

Looks good to me!

vancura commented:
Looks good, but I can't help myself, any idea why "Add Web Reference" has no ellipsis?

mrward commented:
@vancura We can fix up the Add Web Reference.

Should this be run by the UX design board? We now have two different labels for the same action. Add Reference and the existing Edit References. Which both open the Edit References dialog.

vancura commented:
Yes, we need an UXBoard, this is a rather significant change.

It's not the right thing to have two commands with the same action, we should kill one of them. But the problem is that if we kill _Edit References_, it won't be clear how to actually edit references, the _Add Reference_ doesn't tell the story about editing capabilities. It seems we're over-complicating the issue.

What about _Add or Edit References..._ instead of the simple _Add Reference..._, and killing the _Edit References..._?

/c @hbons

ketttY committed:
Change dialog title to "References"

ketttY commented:
Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/975256/

ketttY committed:
[IDE] Drop add prefix for commands under add menu

1. Change "Edit References" to "Add References"
2. Drop the add prefix from the Add sub menu
3. Change "Add files" to "Existing Files"

Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/974919/

mrward commented on main/src/core/MonoDevelop.Ide/ExtensionModel/Commands.addin.xml:
```diff
@@ -239,7 +239,7 @@
 			icon = "md-reference"
-			_label = "_Edit References..."
+			_label = "_Add References..."
 			defaultHandler = "MonoDevelop.Ide.Commands.EditReferencesHandler" />
```
Should this be `_Add Reference...` to be consistent with the other menu which would be `Add -> Reference...`. I think I added the wrong name to the bug report. It would also match VS on Windows //cc @vancura

vancura replied:
Right, singular, I completely agree.

ketttY committed:
references -> reference
